airflow/api_fastapi/execution_api/routes/task_instances.py

```diff
@@ -147,7 +149,26 @@ def ti_update_state(
             next_kwargs=ti_patch_payload.trigger_kwargs,
             trigger_timeout=timeout,
         )
+    elif isinstance(ti_patch_payload, TIRescheduleStatePayload):
+        task_instance = session.scalar(select(TI).where(TI.id == ti_id_str))
```

Comment:
Sqla has a method for getting rows my their primary key -- we should use that here.
```suggestion
        task_instance = session.get(TI, (ti_id_str,))
```

Reply:
Oh nice. Thanks.
Made the change